packages/descendants/src/DescendantsContext.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, Dispatch } from 'react';
+
+import { DescendantsList } from './Descendants.types';
+import { DescendantsReducerAction } from './DescendantsReducer';
+
+export type RegisterDescendantFn<T extends HTMLElement> = (
+  element?: T | null,
+) => () => void;
+
+export interface DescendantsContextProps<T extends HTMLElement> {
+  descendants: DescendantsList<T>;
+  dispatch: Dispatch<DescendantsReducerAction<T>>;
+}
+
+export type DescendantContextType<T extends HTMLElement> = React.Context<
+  DescendantsContextProps<T>
+>;
+
+/**
+ * A factory function that creates a new Context object.
+ * The returned Context object will be used to define which context a descendant is a part of.
+ */
+export const createDescendantsContext = <T extends HTMLElement = HTMLElement>(
+  displayName: string,
+): DescendantContextType<T> => {
+  const context = createContext<DescendantsContextProps<T>>({
+    descendants: [],
+    dispatch: () => {},
+  });
+  context.displayName = displayName;
```

Comment:
is this a React pattern?

Reply:
Which part? Adding the dynamic `displayName`, or the factory function?